solutions/swb-app/src/userRoutes.ts

```diff
@@ -7,39 +7,153 @@
 import {
   CreateRoleSchema,
   CreateUserSchema,
+  Status,
   UpdateRoleSchema,
-  UserManagementService
+  UserManagementService,
+  isUserNotFoundError,
+  isRoleNotFoundError,
+  isInvalidParameterError,
+  isUserAlreadyExistsError,
+  UpdateUserRequest,
+  UpdateUserRequestParser
 } from '@aws/workbench-core-authentication';
+import { validateAndParse } from '@aws/workbench-core-base';
 import Boom from '@hapi/boom';
 import { Request, Response, Router } from 'express';
 import { validate } from 'jsonschema';
 import _ from 'lodash';
 import { wrapAsync } from './errorHandlers';
 import { processValidatorResult } from './validatorHelper';
 
-export function setUpUserRoutes(router: Router, user: UserManagementService): void {
+export function setUpUserRoutes(router: Router, userService: UserManagementService): void {
   router.post(
     '/users',
     wrapAsync(async (req: Request, res: Response) => {
-      processValidatorResult(validate(req.body, CreateUserSchema));
-      const response = await user.createUser(req.body);
-      res.status(201).send(response);
+      try {
+        processValidatorResult(validate(req.body, CreateUserSchema));
```

Comment:
not sure if its out of the scope but can we use zod instead of schema

Reply:
Yea, I'd prefer not to increase scope of this PR, so only zod parser for new code was included. This should be considered as a separate work item